language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java

```diff
@@ -1445,25 +1426,7 @@ private static String getQualifiedModuleName(Module module) {
     }
 
     private static List<String> getBallerinaKeywords() {
-        // NOTE: This is a temporary fix to retrieve lexer defined keywords until we come up with a proper api.
-        // Related discussion can be found in https://github.com/ballerina-platform/ballerina-lang/discussions/28827
-        try {
-            Class<?> aClass = Class.forName("io.ballerina.compiler.internal.parser.LexerTerminals");
-            return Arrays.stream(aClass.getDeclaredFields())
-                    .filter(field -> field.getModifiers() == (Modifier.PUBLIC | Modifier.STATIC | Modifier.FINAL)
-                            && (field.getType() == String.class))
-                    .map(field -> {
-                        try {
-                            return field.get(null).toString();
-                        } catch (IllegalAccessException e) {
-                            return null;
-                        }
-                    })
-                    .filter(Objects::nonNull)
-                    .collect(Collectors.toList());
-        } catch (ClassNotFoundException e) {
-            return Collections.emptyList();
-        }
+        return SyntaxInfo.keywords();
```

Comment:
Any specific reason to maintain this private method?

Reply:
Inside `SyntaxInfo.keywords()`, it dynamically generates the keywords list. Therefore, pre-calculating the list of keywords and having them assigned to `BALLERINA_KEYWORDS` static variable is better for the performance. 

One thing we can do is, instead of using this private method, can do the following:
```
public static final List<String> BALLERINA_KEYWORDS = SyntaxInfo.keywords();
```
and remove the private method. WDYT?

Reply:
Yeah, That was the same idea I was having. This particular method is only used for populating the data. Hence let's move to your suggestion.

Reply:
Done!